syntax = "proto3";

package grpc_server;

service NotificationService {
  rpc SendNotification (NotificationRequest) returns (NotificationResponse) {}
}

message NotificationRequest {
  int32 recipient_id = 1;
  string notification_type = 2;
  string title = 3;
  string message = 4;
  string related_object_type = 5;
  int32 related_object_id = 6;
}

message NotificationResponse {
  bool success = 1;
  string notification_id = 2;
  string message = 3;
}